src/utils/getNameFromPath.js

```diff
@@ -0,0 +1,16 @@
+import getMembers from './getMembers';
+import pascalCase from './pascalCase';
+
+export default function getNameFromPath(path) {
```

Comment:
have you considered just pulling out the literal string value of the LHS here? i don't think we necessarily need to be clever here

Reply:
and then running some sanitization for getting the style node name i guess...

i mean, even with literals below, in principle something wonky could happen

Reply:
what what literal? Most of this matches what react-docgen does so it seemed like a good starting point the main two things i wanted to solve tho that are very common in our codebase is the `export default styled()` case as well as the `Foo.Bar = styled()` case

Reply:
in the `path.isLiteral()` case, if someone does something silly like:

```js
Foo[`B
ar`] = ...
```

(or `Foo['B\nar']`)

bad things may happen

my thinking is just that, if the identifier is `Foo.Bar`, it'd be nice for the `displayName` to be `Foo.Bar` rather than `Foo-Bar`

Reply:
well the display name ends up being `FooBar`, not the pascalCase. I did that instead of `Foo.Bar` because the dots don't work great in a file name: `/file-Foo.Bar.css`